pages/review/[lesson].tsx

```diff
@@ -37,28 +38,25 @@ const Review: React.FC<QueryDataProps<GetAppQuery>> = ({ queryData }) => {
   const context = useContext(GlobalContext)
   const slug = router.query.lesson as string
   const currentLesson = lessons.find(lesson => lesson.slug === slug)
+
   useEffect(() => {
+    redirectUnauthenticated(!session?.user)
```

Comment:
Why did you put this inside useEffect?

Reply:
My understanding is that it's a _side-effect_, so it should be inside `useEffect`. Correct me if I'm wrong.